modules/helfi_etusivu_entities/src/Plugin/Block/SurveyBlock.php

```diff
@@ -0,0 +1,187 @@
+    usort($surveys, static function (NodeInterface $a, NodeInterface $b) {
+      $weightA = $a->get('published_at')->value;
+      $weightB = $b->get('published_at')->value;
+      if ($weightA === $weightB) {
+        return 0;
+      }
+      // More urgent announcements render first.
+      return $weightA < $weightB ? 1 : -1;
+    });
+
+    $referenceField = self::ENTITY_TYPE_FIELDS[$currentEntity?->getEntityTypeId()] ?? NULL;
+
+    foreach ($surveys as $node) {
+      // Check if the announcement should be shown at all pages.
+      if ($node->get('field_survey_content_pages')->isEmpty()) {
+        return $node;
+      }
+
+      // Show survey if current page's entity is found
+      // from the list of referenced entities.
+      if (!empty($referenceField) && $this->hasReference($referenceField, $node, $currentEntity)) {
+        return $node;
+      }
+    }
+
+    return NULL;
+  }
+
+  /**
+   * Loads external survey nodes.
+   *
+   * @return array<int, \Drupal\node\NodeInterface>
+   *   External surveys.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  private function getExternalSurveys(): array {
+    $entityStorage = $this->getExternalEntityStorage('helfi_surveys');
+    $nodes = [];
+
+    /** @var \Drupal\external_entities\ExternalEntityInterface $entity */
+    foreach ($entityStorage->loadMultiple() as $entity) {
+      $linkUrl = NULL;
+      $linkText = NULL;
+      if ($entity->hasField('survey_link_text')) {
+        $linkText = $entity->get('survey_link_text')->value;
+        $linkUrl = $entity->get('survey_link_url')->value;
+      }
+
+      // Create announcement nodes for the block based on external entity data.
+      $nodes[] = Node::create([
+        'uuid' => $entity->get('uuid')->value,
+        'type' => 'survey',
+        'field_survey_link' => ['uri' => $linkUrl, 'title' => $linkText],
+        'langcode' => $entity->get('langcode')->value,
+        'body' => Xss::filter($entity->get('body')->value),
+        'title' => Xss::filter($entity->get('title')->value),
+        'status' => $entity->get('status')->value,
+      ]);
+    }
+
+    return $nodes;
+  }
+
+  /**
+   * Loads local surveys.
+   *
+   * @return array<int, \Drupal\node\NodeInterface>
+   *   Local surveys.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  private function getLocalSurveys() : array {
+    $langcodes = $this->getContentLangcodes();
+
+    $storage = $this->entityTypeManager->getStorage('node');
+
+    // Get all published survey nodes.
+    $query = $storage->getQuery()
+      ->accessCheck(TRUE)
+      ->condition('type', 'survey')
+      ->condition('status', NodeInterface::PUBLISHED)
+      ->condition('langcode', $langcodes, 'IN')
+      ->sort('published_at', 'DESC');
```

Comment:
Some instances (such as KYMP) don't have the `published_at` column available, which causes an error here.

Reply:
Hmm. Platform config requires `publication_date` [module](https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/main/composer.json#L48). I'll look into this.

Reply:
It's also listed as a dependency on multiple modules: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/main/modules/helfi_node_page/helfi_node_page.info.yml#L8, https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/main/modules/helfi_node_announcement/helfi_node_announcement.info.yml#L10 and https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/main/modules/helfi_platform_config_base/helfi_platform_config_base.info.yml#L37.

Maybe those dependencies were added after the modules had been installed on KYMP?

Reply:
Should be fixed by: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/773/commits/71806ee25b693079c7f77d51815f4635a1b04f27.